backend/src/repositories/BaseRepository.js

```diff
@@ -11,6 +11,10 @@ class BaseRepository {
     return this.model.findById(id);
   }
 
+  async findByEmail(email) {
+    return this.model.findOne({ email });
+  }
+
```

Comment:
Suggestion: `findbyEmail` is a user specific function I would suggest you to keep it in the user repository. Or if we want to remove it, we can have a generic function here like `findByFilter` or `findOne` to pass a variable (filter) like email or id for item to fine one item. 

In #7 I have already made changes to BaseRepository code. Kindly check that PR refactor code.